source/diffHandler.py

```diff
@@ -21,25 +22,18 @@ class DiffAlgo(AutoPropertyObject):
 	def diff(self, newText: str, oldText: str) -> List[str]:
 		raise NotImplementedError
 
-	_abstract_unit = True
-
-	def _get_unit(self):
+	@abstractmethod
+	def _getText(self, ti: TextInfo) -> str:
 		raise NotImplementedError
 
-	def _get_name(self):
-		return "diffing algorithm"
-
-	def __repr__(self):
-		return f"{self.unit}-based ({self.name})"
-
 
 class DiffMatchPatch(DiffAlgo):
-	name = "diff match patch"
-	unit = UNIT_CHARACTER
-
-	def __init__(self, *args, **kwargs):
-		self._proc = None
-		return super().__init__(*args, **kwargs)
+	(
+		"A character-based diffing approach, using the Google Diff Match Patch "
+		"library in a proxy process (to work around a licence conflict)."
+	)
+	_proc = None
+	_lock = Lock()
 
 	def _initialize(self):
 		if not self._proc:

```

Comment:
I think this should be `DiffMatchPatch._proc` same with lock. Comments on the lock would also be helpful to explain that you don't want multiple threads trying to get a diff from the single process in an interleaved way.